File: WebApplication1/Contracts/IStudentsRepository.cs
using ApiDtoLibrary.Students;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using WebApplication1.Contracts;
using WebApplication1.Extensions;
using WebApplication1.Services.People;

namespace WebApplication1.Contracts
{
    public interface IStudentsRepository : IGenericRepository<Student, GetStudent>
    {
        List<Student> SortFilterStudents(StudentOrderByOptions orderByOption, StudentFilterByOptions filterByOption, string filter);
        Task RemoveStudentCourseAsync(Guid studentId, Guid courseId);
        Task<Student> AddStudentWithCoursesAsync(Student updatedStudent, IEnumerable<Guid> coursesIds);
        Task<Guid> UpdateStudentWithCoursesAsync(Student updatedStudent, IEnumerable<Guid> coursesIds);
        Task<bool> IndexIsOccupied(string index);
    }
}

File: FsUniversityApi/Contracts/IStudentsRepository.fs
namespace FsUniversityApi.Contracts

open FsUniversityApi.Database.Entities.StudentCourseProfessor
open System.Threading.Tasks
open System
open System.Collections.Generic

type IStudentsRepository =
    abstract member UpdateWithCoursesAsync : Student * IEnumerable<Guid> -> Task<Guid>
    abstract member IndexIsOccupied : string -> Task<bool>
    abstract member AddWithCoursesAsync : Student * IEnumerable<Guid> -> Task<Student>
    abstract member DeleteStudentsCourseAsync : Guid * Guid -> Task